src/components/dataset/query/sidebar/filter/FreetextFilter.jsx

```diff
@@ -43,20 +43,11 @@ export class FreetextFilter extends React.PureComponent {
     }
   };
 
-  onChange = (event) => {
+  onPaste = (event) => {
     const { handleChange } = this.props;
-    const { value } = event.target;
-    let selections;
-    if (value.includes(',')) {
-      selections = value.split(',');
-    } else if (value.includes(' ')) {
-      selections = value.split(' ');
-    } else {
-      return;
-    }
-    const trimmed = _.map(selections, _.trim);
-    const nonEmpty = _.filter(trimmed, (t) => t !== '');
-    handleChange(nonEmpty);
+    event.preventDefault();
+    const text = event.clipboardData.getData('text');
+    handleChange(text.split(/[ ,\n]+/));
```

Comment:
I *think* you still need to filter out the empty strings (like you did in the 'nonEmpty' variable above).

Reply:
+1, might be good to pull this regex out to a const and give it a name so it's a bit more readable as well